src/Sarif/Core/PropertyBagHolder.cs

```diff
@@ -180,5 +180,18 @@ public void RemoveProperty(string propertyName)
 
         [JsonIgnore]
         public TagsCollection Tags { get; }
+
+        public virtual bool ShouldSerializeProperties()
+        {
+            return PropertyBagHasAtLeastOneNonNullValue();
+        }   
+
+        public bool PropertyBagHasAtLeastOneNonNullValue()
+        {
+            bool hasAtLeastOneNonNullTag = this.Tags.Where(t => !string.IsNullOrEmpty(t)).Count() > 0;
```

Comment:
>. [](start = 52, length = 1)

Missing null check.

Reply:
Tags is always non-null, check out the default ctor.

---
In reply to: [229092005](https://github.com/Microsoft/sarif-sdk/pull/1121#discussion_r229092005) [](ancestors = 229092005)

Reply:
ok.

---
In reply to: [229102696](https://github.com/Microsoft/sarif-sdk/pull/1121#discussion_r229102696) [](ancestors = 229102696,229092005)